gobblin-runtime/src/main/java/gobblin/runtime/TaskContext.java

```diff
@@ -132,6 +132,9 @@ public WriterOutputFormat getWriterOutputFormat(int branches, int index) {
   public List<Converter<?, ?, ?, ?>> getConverters(int index) {
     String converterClassKey =
         ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.CONVERTER_CLASSES_KEY, index);
+
+    this.workUnitState.setProp(ConfigurationKeys.FORK_BRANCH_ID_KEY, index);
```

Comment:
Since this method is called in every `Fork`s of a `Task` and soon `Fork`s will run concurrently, there will be race conditions on this key.